low-level/files/opt/seagate/sspl/setup/sspl_test.py

```diff
@@ -25,17 +33,47 @@ class SSPLTestCmd:
     def __init__(self, args: list):
         self.args = args
         self.name = "sspl_test"
+        self.plan = "self_primary"
+        self.avoid_rmq = False
 
     def process(self):
-        # TODO: Need to convert run_tests.sh from shell to python
-        test_plan=None
-        sspl_test_plans = ["sanity", "alerts", "self_primary","self_secondary"]
-        for i in range(len(self.args)):
-            if self.args[i] in sspl_test_plans:
-                test_plan = self.args[i]
-        if test_plan is None:
-            test_plan="self_primary"
-        CMD = f"{TEST_DIR}/run_tests.sh test {test_plan}"
+        global_config = self.args.config[0]
+        self.plan = self.args.plan[0]
+        self.avoid_rmq = self.args.avoid_rmq
+
+        Conf.load("sspl", sspl_config_path)
+        Conf.load("sspl_test", sspl_test_config_path)
+
+        # Take back up of sspl test config
+        sspl_test_backup = '/etc/sspl_tests.conf.back'
+        shutil.copyfile(sspl_test_file_path, sspl_test_backup)
+        global_config_url = Conf.get("sspl", "SYSTEM_INFORMATION>global_config_url")
+
+        # Global config path in sspl_tests.conf will be referred by sspl_tests later
+        Conf.copy("global_config", "sspl_test")
```

Comment:
This is creating global config copy right, can we name it accordingly, currently its confusing with the name, like sspl_tests_global_config ?

Also where this index 'global_config' initalized? One arg coming with same name, though how a confstor handle for that index would come as arg?

Reply:
Conf object is created once we loaded config with index. This is a Singleton and throwing exception if we create duplicate index.

`where this index 'global_config' initalized?` global_config is already loaded in sspl_setup under TestCmd.validate().

`Conf.copy("global_config", "sspl_test")`
After loading sspl_test config, we copy global_config to sspl_test config. Now sspl_test config will have both global_config and test config.

I will add proper comment before this.

Reply:
alright, thanks. Yes code comments will help